AI Toolkit doc updates for v0.7.0 (#11870)
flutter/website: src/content/ai-toolkit/feature-integration.md

* **Welcome messages**: Display an initial greeting to users.
* **Suggested prompts**: Offer users predefined prompts to guide interactions.
* **System instructions**: Provide the LLM with specific input to influence its responses.

> * **Disabling attachments and audio input**: Removing optional parts of the chat UI.
> * **Manageing cancel or error behavior**: Changing the user cancelation or LLM error behavior.

Reviewer: ```suggestion
* **Disable attachments and audio input**: Remove optional parts of the chat UI.
* **Manage cancel or error behavior**: Change the user cancellation or LLM error behavior.
```

> * **Managing history**: Every LLM provider allows for managing chat history,

Reviewer: ```suggestion
* **Manage history**: Every LLM provider allows for managing chat history,
```

}
```

> ## Disabling attachments and audio input

Reviewer: ```suggestion
## Disable attachments and audio input
```

> If you'd like to disable attachments (the + button) or audio input (the mic button),

Reviewer: ```suggestion
If you'd like to disable attachments (the **+** button) or audio input (the mic button),
```

you can do so with the `enableAttachments` and `enableVoiceNotes` parameters to
the `LlmChatView` constructor:

```dart
class ChatPage extends StatelessWidget {
  const ChatPage({super.key});

  @override
  Widget build(BuildContext context) {
    ...

    return Scaffold(
      appBar: AppBar(title: const Text('Restricted Chat')),
      body: LlmChatView(
        ...
        enableAttachments: false,
        enableVoiceNotes: false,
      ),
    );
  }
}
```

Both of these flags default to `true`.

> ## Manageing cancel or error behavior

Reviewer: ```suggestion
## Manage cancel or error behavior
```

Be default, when the use cancels an LLM request, the LLM's response will be
appended with the string "CANCEL" and a message will pop up that the user has
canceled the request. Likewise, in the event of an LLM error, like a dropped
network connection, the LLM's response will be appended with the
string "ERROR" and an alert dialog will pop up with the details of the error.

You can override the cancel and error behavior with the `cancelMessage`,
`errorMessage`, `onCancelCallback` and `onErrorCallback` parameters of the

> `LlmChatView`. For example, the following code replaces the default cancelation

Reviewer: ```suggestion
`LlmChatView`. For example, the following code replaces the default cancellation
```

handling behavior:

```dart
class ChatPage extends StatelessWidget {
  ...

  void _onCancel(BuildContext context) {
    ScaffoldMessenger.of(
      context,
    ).showSnackBar(const SnackBar(content: Text('Chat cancelled')));
  }

  @override
  Widget build(BuildContext context) => Scaffold(
    appBar: AppBar(title: const Text(App.title)),
    body: LlmChatView(
      ...
      onCancelCallback: _onCancel,
      cancelMessage: 'Request cancelled',
    ),
  );
}
```

You can override any or all of these parameters and the `LlmChatView` will use
its defaults for anything you don't override.

> ## Managing history

Reviewer: ```suggestion
## Manage history
```

}
```

## Disable attachments and audio input

If you'd like to disable attachments (the **+** button) or audio input (the mic button),
you can do so with the `enableAttachments` and `enableVoiceNotes` parameters to
the `LlmChatView` constructor:

```dart
class ChatPage extends StatelessWidget {
  const ChatPage({super.key});

  @override
  Widget build(BuildContext context) {
    ...

    return Scaffold(
      appBar: AppBar(title: const Text('Restricted Chat')),
      body: LlmChatView(
        ...
        enableAttachments: false,
        enableVoiceNotes: false,
      ),
    );
  }
}
```

Both of these flags default to `true`.

## Manage cancel or error behavior

> Be default, when the use cancels an LLM request, the LLM's response will be

Reviewer: ```suggestion
Be default, when the user cancels an LLM request, the LLM's response will be
```
